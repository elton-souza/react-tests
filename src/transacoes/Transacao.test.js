import React from 'react';
import { render } from '@testing-library/react';
import Transacao from './Transacao';
import Transacoes from './Transacoes';

describe('Componente de transação do extrato', () => {
    it('O snapshot do componente deve permanecer sempre o mesmo', () => {

        const { container } = render(<Transacao data='08/06/2022' tipo='saque' valor='100' />)

        expect(container.firstChild).toMatchSnapshot()
    })
})

describe('Componente de lista de transações do extrato', () => {
    it('O Snapshot do componente deve permanecer sempre renderizar uma lista', () => {
        const dados = [{
            data: '08/06/2022',
            tipo: 'saque',
            valor: '100'
        },
        {
            data: '09/06/2022',
            tipo: 'deposito',
            valor: '150'
        }]
        const { container } = render(<Transacoes transacoes={dados} />)

        expect(container.firstChild).toMatchSnapshot();
    })
})