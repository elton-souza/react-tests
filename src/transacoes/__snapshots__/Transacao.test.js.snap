// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente de lista de transações do extrato O Snapshot do componente deve permanecer sempre renderizar uma lista 1`] = `
<div
  data-testid="transacoes"
>
  <div
    class="transacao-container"
  >
    <p>
      08/06/2022
    </p>
    <p />
    <p>
      R$ 
      100
    </p>
  </div>
  <div
    class="transacao-container"
  >
    <p>
      09/06/2022
    </p>
    <p />
    <p>
      R$ 
      150
    </p>
  </div>
</div>
`;

exports[`Componente de transação do extrato O snapshot do componente deve permanecer sempre o mesmo 1`] = `
<div
  class="transacao-container"
>
  <p>
    08/06/2022
  </p>
  <p>
    saque
  </p>
  <p>
    R$ 
    100
  </p>
</div>
`;
